examples/nucleo_g474re/sx128x_lora/main.cpp

```diff
@@ -0,0 +1,277 @@
+    codingRate : modm::sx128x::LoRa::ModulationParams::CodingRate::Cr_Li_4_7
+};
+
+static constexpr modm::sx128x::LoRa::PacketParams packetParams = {
+    preambleLength : 12,
+    headerType : modm::sx128x::LoRa::PacketParams::HeaderType::Explicit,
+    payloadLength : 4,
+    crc : modm::sx128x::LoRa::PacketParams::Crc::Enable,
+    invertIq : modm::sx128x::LoRa::PacketParams::InvertIq::Standard
+};
+
+}
+
+class RxThread : public modm::sx128x, public modm::pt::Protothread
+{
+    using Reset = GpioB3;
+    using Busy = GpioB4;
+    using Dio1 = GpioB5;
+    using Dio2 = GpioB6;
+    using Dio3 = GpioB7;
+
+    using Nss = GpioD2;
+    using Transport = modm::Sx128xTransportSpi<SpiMaster, Nss>;
+
+public:
+    RxThread() {}
+
+    inline bool
+    run()
+    {
+        PT_BEGIN();
+
+        Nss::setOutput(modm::Gpio::High);
+        Reset::setOutput(modm::Gpio::Low);
+        Busy::setInput(modm::platform::Gpio::InputType::PullDown);
+
+        Dio1::setInput(modm::platform::Gpio::InputType::PullDown);
+        Exti::connect<Dio1>(Exti::Trigger::RisingEdge, [](uint8_t) { rx::dio1 = true; });
+
+        Dio2::setInput(modm::platform::Gpio::InputType::PullDown);
+        Exti::connect<Dio2>(Exti::Trigger::RisingEdge, [](uint8_t) { rx::dio2 = true; });
+
+        Dio3::setInput(modm::platform::Gpio::InputType::PullDown);
+        Exti::connect<Dio3>(Exti::Trigger::RisingEdge, [](uint8_t) { rx::dio3 = true; });
+
+        PT_CALL(radio.reset());
+        PT_CALL(radio.setStandby());
+
+        // Initialize the sx128x
+        PT_CALL(radio.setPacketType(PacketType::LoRa));
+        PT_CALL(radio.setRfFrequency(2457_MHz / radio.frequencyLsb));
+        PT_CALL(radio.setRegulatorMode(RegulatorMode::Ldo));
+        PT_CALL(radio.setBufferBaseAddress(0, 0));
+        PT_CALL(radio.setModulationParams(modulationParams));
+        PT_CALL(radio.writeRegister(Register::SfAdditionalConfiguration, 0x32));
+        PT_CALL(radio.writeRegister(Register::FrequencyErrorCorrection, 0x01));
+        PT_CALL(radio.setPacketParams(packetParams));
+        PT_CALL(radio.setDioIrqParams(Irq::RxDone | Irq::RxTxTimeout, Irq::RxDone, Irq::RxTxTimeout));
+        PT_CALL(radio.setRx(sx128x::PeriodBase::ms1, 1000));
+
+        MODM_LOG_DEBUG << "Sx128x initialization complete!" << modm::endl;
+
+        while (true)
+        {
+            if (rx::dio1.exchange(false))
+            {
+                PT_CALL(radio.getIrqStatus(&irqStatus));
+                if (irqStatus.any(Irq::RxDone))
+                {
+                    PT_CALL(radio.clearIrqStatus(Irq::RxDone | Irq::RxTxTimeout));
+                    PT_CALL(radio.setRx(sx128x::PeriodBase::ms1, 1000));
+
+                    PT_CALL(radio.getRxBufferStatus(&rxBufferStatus));
+                    PT_CALL(radio.getPacketStatus(&packetStatus));
+                    PT_CALL(radio.readBuffer(rxBufferStatus.rxStartBufferPointer, std::span{buffer, rxBufferStatus.rxPayloadLength}));
```

Comment:
Is it guaranteed `rxBufferStatus.rxPayloadLength` is never larger than `bufferSize`?

Reply:
I checked that the transceiver is equipped with a 256 byte RAM data buffer, so I will change the buffer size in the thread to be the same. Then there should be no problem.